src/olympia/addons/tests/test_models.py

```diff
@@ -1689,6 +1695,79 @@ def test_promoted_groups(self):
         assert not addon.promoted_groups()
         assert addon.promoted_groups(currently_approved=False)
 
+    def test_multiple_promoted_groups(self):
```

Comment:
This should be split into several tests. Long tests that test multiple behaviors like this have a few issues:
- they take a long time to read and understand, really easy to miss something
- they give poor signal when something fails. Assume the first assertion fails, you lose all info about the rest of the test. A lot less granular than multiple specific tests
- they are hard to maintain. Mostly because of the first point, but also because you have a lot of sequentially ordered logic and when the code changes it may or may not continue making sense in the order you have. Isolated tests are harder to break because they are sensitive to smaller parts of the code base.